### PR TITLE
Hide empty categories and categories that include only hidden channels

### LINKED ISSUE #19885: Hide private channel if created with the "hidden option"
# Feature Issue

### User Story

As a community admin
I want to be able to hide private channel from those who don't have access
So that I don't disclose private information about my community

### Description

Private channel of a community can be set so that they are hidden when adding permissions/creating the channel.
Those channels should not be shown to users who don't have access. If a category has no visible channels, it should not be shown.

### PR DESCRIPTION
**Pull request by Parveshdhull**
fixes https://github.com/status-im/status-mobile/issues/19885

status: ready

### CODE REVIEW

File: src/status_im/contexts/communities/overview/view.cljs [hunk 1]
```diff
@@ -73,25 +73,26 @@
...
-         name])
-      (when-not collapsed?
-        [rn/view {:style {:padding-horizontal 8}}
-         (let [last-item-index (dec (count chats))]
-           (map-indexed
-            (fn [index chat]
-              ^{:key (:id chat)}
-              [channel-chat-item community-id chat (= index last-item-index)])
-            chats))])])])
+     (when (seq chats)
```
Comment (flexsurfer) [inline src/status_im/contexts/communities/overview/view.cljs:76]: `seq?`

Comment (Parveshdhull) [reply src/status_im/contexts/communities/overview/view.cljs:76]: Thank you @flexsurfer for review.

Guideline suggests seq: https://github.com/bbatsov/clojure-style-guide#nil-punning

Comment (Parveshdhull) [reply src/status_im/contexts/communities/overview/view.cljs:76]: And [seq?](https://clojuredocs.org/clojure.core/seq_q) is just for checking if passed argument implements ISeq

Comment (flexsurfer) [reply src/status_im/contexts/communities/overview/view.cljs:76]: it depends if you are checking on nil or on empty?

Comment (Parveshdhull) [reply src/status_im/contexts/communities/overview/view.cljs:76]: for nil check I guess, I could have used chats directly 😅, but here I am checking for empty. (Don't know if returned object is empty or nil, so empty seems safer)

Comment (flexsurfer) [reply src/status_im/contexts/communities/overview/view.cljs:76]: anyway here should be seq i guess

**Review by flexsurfer — APPROVED** [review]

File: src/status_im/contexts/communities/overview/view.cljs [hunk 2]
```diff
@@ -379,15 +380,17 @@
...
+      {:title      (i18n/label :t/community-overview)
        :type       :title
        :text-align :left
        :icon-name  :i/close
        :on-press   #(rf/dispatch [:navigate-back])}]
      [quo/empty-state
       {:image           (resources/get-themed-image :cat-in-box theme)
        :description     (when-not fetching? (i18n/label :t/here-is-a-cat-in-a-box-instead))
-       :title           (if fetching? "Fetching community..." "Failed to fetch community")
+       :title           (if fetching?
```
Comment (Parveshdhull) [inline src/status_im/contexts/communities/overview/view.cljs:391]: Small additional change, leftover from https://github.com/status-im/status-mobile/pull/19968#discussion_r1598790387

(should not affect review or e2e, just moved string to translation files)

cc @ajayesivan @seanstrom

**Review by ajayesivan — APPROVED** [review]

**Review by BalogunofAfrica — APPROVED** [review]

**Comment by qoqobolo** [comment]
Thanks for your work @Parveshdhull!
LGTM, PR can be merged.

**Comment by Parveshdhull** [comment]
Not sure why unit-tests are failing after rebase. @siddarthkay Please can you check.

**Comment by Parveshdhull** [comment]
ok, waiting if works this time

**Comment by Parveshdhull** [comment]
> Not sure why unit-tests are failing after rebase. @siddarthkay Please can you check.

Maybe because I pressed restart tests button, but now it seems all tests are failing

**Comment by siddarthkay** [comment]
Hey @Parveshdhull : will take a look ASAP

**Comment by Parveshdhull** [comment]
> Hey @Parveshdhull : will take a look ASAP

awsm, thank you very much

**Comment by Parveshdhull** [comment]
https://github.com/status-im/status-mobile/pull/19994 PR merged now, without any issues :thinking:

**Comment by siddarthkay** [comment]
https://github.com/status-im/status-mobile/pull/20061 also does not have the same issue. so probably this issue is host related. `linux-03`

**Comment by Parveshdhull** [comment]
> #20061 also does not have the same issue. so probably this issue is host related. `linux-03`

hi, reverting https://github.com/status-im/status-mobile/pull/20044, did fix tests for my PR
(related [conversation](https://discord.com/channels/1210237582470807632/1217173176086302821/1240617733821960282))

**Comment by siddarthkay** [comment]
Hi @Parveshdhull 
feel free to merge this PR I'll make a follow up PR and investigate.
Thank you

**Comment by Parveshdhull** [comment]
> Hi @Parveshdhull feel free to merge this PR I'll make a follow up PR and investigate. Thank you

cool, thank you

**Pull request merged**